Marlin/src/lcd/menu/menu_tramming.cpp

```diff
@@ -50,32 +52,32 @@ static bool probe_single_point() {
   do_blocking_move_to_z(TERN(BLTOUCH, Z_CLEARANCE_DEPLOY_PROBE, Z_CLEARANCE_BETWEEN_PROBES));
   // Stow after each point with BLTouch "HIGH SPEED" mode for push-pin safety
   const float z_probed_height = probe.probe_at_point(tramming_points[tram_index], TERN(BLTOUCH_HS_MODE, PROBE_PT_STOW, PROBE_PT_RAISE), 0, true);
-  DEBUG_ECHOLNPAIR("probe_single_point: ", z_probed_height, "mm");
   z_measured[tram_index] = z_probed_height;
+  if (reference_index < 0) reference_index = tram_index;
   move_to_tramming_wait_pos();
 
-  return !isnan(z_probed_height);
+  DEBUG_ECHOLNPAIR("probe_single_point(", tram_index, ") = ", z_probed_height, "mm");
+  return (z_isvalid[tram_index] = !isnan(z_probed_height));
 }
 
-static void _menu_single_probe(const uint8_t point) {
-  tram_index = point;
-  DEBUG_ECHOLNPAIR("Screen: single probe screen Arg:", point);
+static void _menu_single_probe() {
+  tram_index = MenuItemBase::itemIndex;
```

Comment:
The tricky part here is that if `_menu_single_probe` were to contain any of the menu item macros that write to this index, it and `tram_index` will get trounced, and a future reviewer may not realize this is why `tram_index` is getting stepped on. Something to watch out for.